test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,493 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
+      }).where({name: 4}).get()
       ).to.be.fulfilled
     );
 
-    it('should query with index', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product,
-        indexName: 'Testing_Sort',
-        waitForNonStaleResults: true
-      }).where({
-        uid: [4, 6, 90]
-      }).get();
+    it('should fail query with non-existing index', async () => {
+      const results = await store.openSession().query<Universal>({
+        indexName: 'none'
+      }).whereIn('Tag', 'Products').get();
+
+      expect(results).to.have.lengthOf(0);
+
+    });
+
+    it('should query by @all_docs index', async () => {
+      const results: Universal[] = await store.openSession().query<Universal>({
+        indexName: 'Universals'
+      }).whereIn('name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function (obj) {
+        expect(obj.name).to.equal('withNesting');
+      });
+
+    });
+
+    it('should paginate', async () => {
+      const pageSize: number = 2;
+
+      const totalCount: Universal[] = await store.openSession().query<Universal>({
+        indexName: 'Universals',
+        WaitForNonStaleResults: true
+      }).whereIn('name', 'withNesting').get();
+
+      const totalPages: number = Math.ceil(totalCount.length / pageSize);
+
+      totalCount.map(function (obj) {
+        expect(obj.name).to.equal('withNesting');
+      });
+
+      expect(totalCount).to.have.lengthOf(10);
+      expect(totalPages).to.equals(5);
 
-      expect(results).to.have.lengthOf(3);
     });
 
-    it('should query by between', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product,
-        indexName: 'Testing_Sort',
-        waitForNonStaleResults: true
-      }).whereBetween<number>('uid', 2, 4).get();
-      
-      expect(results).to.have.lengthOf(1);
+    it('should query with whereEquals', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereEquals('name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function (obj) {
+        expect(obj.name).to.equal('withNesting');
+      });
+
     });
 
-    it('should query by between or equal', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product,
-        indexName: 'Testing_Sort',
-        waitForNonStaleResults: true
-      }).whereBetweenOrEqual<number>('uid', 2, 4).get();
-      
-      expect(results).to.have.lengthOf(3);
+    it('should query with andAlso', async () => {
+      const results = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .andAlso();
```

Comment:
I believe you're missing something past the `andAlso()` call. Left like this it does not affect the query at all.